processor/ratelimitprocessor/ratelimiter.go

```diff
@@ -51,6 +51,20 @@ type RateLimiter interface {
 // high cardinality: tenant ID would be a good choice. For rate
 // limiting by IP (e.g. to avoid DDoS), consider running OpenTelemetry
 // Collector behind a WAF/API Gateway/proxy.
+
+const (
+	ProjectID = "x-elastic-project-id"
+)
+
+func getProjectIDFromMetadata(ctx context.Context) string {
```

Comment:
We wanted to upstream this component at some point, so its better to abstract this out and add all keys passed via `metadata_keys` as attributes to the telemetry metrics.

Reply:
I added https://github.com/elastic/opentelemetry-collector-components/pull/562/files#diff-03ffdee6e5e78b42afe5d4f984f4bf495b5b10877d11593800c0a30abbd4e884R60 

WDYT?